AndroidSDKCore/src/main/java/com/leanplum/internal/CountAggregator.java

```diff
@@ -0,0 +1,62 @@
+package com.leanplum.internal;
+
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CountAggregator {
+
+    public static final CountAggregator INSTANCE = new CountAggregator();
+
+    private HashSet<String> enabledCounters = new HashSet<>();
+    private final HashMap<String, Integer> counts = new HashMap<>();
+
+    public void setEnabledCounters(HashSet<String> enabledCounters) {
+        this.enabledCounters = enabledCounters;
+    }
+
+    public void incrementCount(String name) {
+        incrementCount(name, 1);
+    }
+
+    public void incrementCount(String name, int incrementCount) {
+        if (enabledCounters.contains(name)) {
+            Integer count = 0;
+            if (counts.containsKey(name)) {
+                count = counts.get(name);
+            }
+            count = count + incrementCount;
+            counts.put(name, count);
+        }
+    }
+
+    private HashMap<String, Integer> getAndClearCounts() {
+        HashMap<String, Integer> previousCounts = new HashMap<>();
+        previousCounts.putAll(counts);
+        counts.clear();
+        return previousCounts;
+    }
+
+    public void sendAllCounts() {
```

Comment:
Can we test this? If not, why? Ideally we could mock Request and test that it receives the correct API calls.

Reply:
we will add a test for this in the future